src/api/plugins/tdbp/warren_tdbp/indicators.py

```diff
@@ -0,0 +1,259 @@
+            # Filter on statements emitted within the floating window
+            window_statements = statements[since <= statements["timestamp"]]
+
+            # Count unique actors active in the floating window
+            temp_dynamic_cohort = window_statements["actor.account.name"].unique()
+            temp_dynamic_cohort_size = len(temp_dynamic_cohort)
+
+            # Loop on actions
+            actions = (
+                window_statements.groupby(["object.id"])["actor.account.name"]
+                .nunique()
+                .reset_index()
+            )
+
+            # Find active actions on the current window
+            for idx in actions.index:
+                action_id = actions.iloc[idx]["object.id"]
+                action_cohort = actions.iloc[idx]["actor.account.name"]
+
+                if (
+                    0.1 * temp_dynamic_cohort_size <= action_cohort
+                    and action_cohort >= self.dynamic_cohort_size
+                ):
+                    active_actions_iris.append(action_id)
+
+            if len(active_actions_iris) < self.active_actions_min:
+                since -= timedelta(days=1)  # step back from one day
+            else:
+                self.sliding_window = DatetimeRange(since=since, until=self.until)
+                self.dynamic_cohort = temp_dynamic_cohort
+                self.active_actions = self._fetch_active_actions_context(
+                    active_actions_iris, temp_dynamic_cohort_size
+                )
+
+                return SlidingWindow.construct(
+                    **{
+                        "window": self.sliding_window,
+                        "active_actions": self.active_actions,
+                        "dynamic_cohort": self.dynamic_cohort,
+                    }
+                )
+
+        return 0
+
+    def _fetch_active_actions_context(
+        self, statements, iris, dynamic_cohort_size: int
+    ) -> None:
+        """Retrieves contextual information about active actions."""
+        active_actions: List[ActiveAction] = []
+
+        for iri in iris:
+            activation_date = min(
+                statements.loc[(statements["object.id"] == iri)]["timestamp"]
+            )
+            type = statements.loc[(statements["object.id"] == iri)][
+                "context.extensions.http://lrs.learninglocker.net/define/extensions/info.event_name"
+            ].unique()
+
+            cohort = statements.loc[(statements["object.id"] == iri)][
+                "actor.account.name"
+            ].unique()
+
+            score = len(cohort) / dynamic_cohort_size
+
+            active_actions.append(
+                ActiveAction(
+                    id=iri,
+                    activation_date=activation_date,
+                    type=type,
+                    cohort=cohort,
+                    score=score,
+                )
+            )
+
+        return active_actions
+
+
+class ScoreIndicator(SlidingWindowIndicator):
```

Comment:
IMO it should be derived from the base indicator and use the `SlidingWindowIndicator` in its `compute`.

```suggestion
class ScoreIndicator(BaseIndicator):
```